packages/flutter_migrate/lib/src/migrate_logger.dart

```diff
@@ -0,0 +1,73 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'base/logger.dart';
+import 'base/terminal.dart';
+
+const int kDefaultStatusIndent = 2;
+
+class MigrateLogger {
+  MigrateLogger({
+    required this.logger,
+    this.verbose = false,
+    this.silent = false,
+  });
+
+  final Logger logger;
+  // We keep a spinner going and print periodic progress messages
+  // to assure the developer that the command is still working due to
+  // the long expected runtime.
+  Status? status;
```

Comment:
Why is this nullable? I'm not clear from reading this code what the distinction is between "never started" (null) and "started, but then stopped (non-null).

Reply:
I'll change this to initialize the spinner in the constructor